u19_pipeline/automatic_job/ephys_element_populate.py

```diff
@@ -22,12 +22,22 @@ def run(display_progress=True, reserve_jobs=False, suppress_errors=False):
         precluster_param_list_id = (recording_process.Processing.EphysParams & 
                                                 key).fetch1('precluster_param_list_id')
         
+        precluster_paramsets = (ephys_element.PreClusterParamList.ParamOrder() & 
+                                dict(precluster_param_list_id=precluster_param_list_id)
+                               ).fetch('paramset_idx')
+
+        if len(precluster_paramsets)==0 or \
+           (len(precluster_paramsets)==1 and precluster_paramsets[0]==None):
```

Comment:
Summarizing: "If there are no precluster paramsets, OR if the pre is only one item and that item is `None`" - correct?
My gut says that a 'order of operations' table should have an integrity check to prevent the latter case, but the only was I can imagine to accomplish that is a helper method for inserting in the table with that assertion baked in, like we do for paramset_id entry

Reply:
Thanks @CBroz1.  Yes, your summary is correct.

I don't have a strong opinion here, but perhaps we want the latter case so that we explicitly keep track that no pre-clustering was performed?

Reply:
makes sense